src/deluge/dsp/delay/delay_buffer.h

```diff
@@ -19,53 +19,65 @@
 
 #include "dsp/stereo_sample.h"
 #include <cstdint>
+#include <expected>
 
 class StereoSample;
 
-#define delaySpaceBetweenReadAndWrite 20
-#define DELAY_BUFFER_MAX_SIZE 88200
-#define DELAY_BUFFER_MIN_SIZE 1
-#define DELAY_BUFFER_NEUTRAL_SIZE 16384
-
-struct DelayBufferSetup {
-	int32_t actualSpinRate;           // 1 is represented as 16777216
-	int32_t spinRateForSpedUpWriting; // Normally the same as actualSpinRate, but subject to some limits for safety
-	uint32_t divideByRate;            // 1 is represented as 65536
-	int32_t rateMultiple;
-	uint32_t writeSizeAdjustment;
-};
+constexpr ptrdiff_t delaySpaceBetweenReadAndWrite = 20;
 
 class DelayBuffer {
 public:
-	DelayBuffer();
-	~DelayBuffer();
+	constexpr static size_t kMaxSize = 88200;
+	constexpr static size_t kMinSize = 1;
+	constexpr static size_t kNeutralSize = 16384;
+
+	struct Config {
```

Comment:
I think this should stay out of the class, or be set seperately and not passed as an argument to the functions that need it. It's misleading that the delaybuffer has a config but the functions need to be passed a config as well

Reply:
I've refactored this so that it's clear this is a _resampling_ config. The buffer no longer has intrinsic knowledge of if it's resampling or not, instead the difference between buffer write depends on the presence of the config (as an optional)

Reply:
This is clearer, but as far as I can tell there's still nothing that actually uses the config inside the delay class, they all refer to a passed in config instead - I think it should just be on its own in this file

Reply:
Disregard - misread